oscarlab/graphene
# Migrate and remove test/apps submodule

## Description of the changes 

We decided to merge the [sample app integrations](https://github.com/oscarlab/graphene-tests) submodule back because working with git submodules turned out to be really painful. The only blocker for this was the fact that previously, it contained a lot of binary blobs and copy-pasted sources, but this was cleaned up recently.

Application integrations were moved to `Examples` directory and `LTP` was moved to `LibOS/shim/test/ltp` (the old location didn't make too much sense).

The first commit only copies everything from the submodule into this repository, the next one fixes (hopefully) all required stuff to make it work in the new location. I think it's better to leave both commits unsquashed in the final version, but that's open for discussion.

## Migration plan

1. Merge this PR
2. Merge https://github.com/oscarlab/graphene-tests/pull/79 (deprecation notice in the README)
3. Archive [graphene-tests](https://github.com/oscarlab/graphene-tests)
4. Migrate PRs which we still want to merge from [graphene-tests/pulls](https://github.com/oscarlab/graphene-tests/pulls)

## How to review this PR?

* Verify that I didn't miss anything from the original repo.
* Check if you're ok with the credits in the commit message in the current form and whether they are correct.
* There are 3 commits which should be reviewed separately and I already set Reviewable for this PR to commit-by-commit mode. I needed to fix up some things and rebase the PR in the meantime, so the diff ranges you should review are:
   * First commit: base -> r3. This commit is best reviewed locally, by just copying the old repo and diffing.
   * Second commit: r3 -> r4.
   * Third commit: r4 -> r6 (I needed to push a small fixup for this one).

## How to test this PR?

* Check if all apps build and run correctly (I may have missed some path fixups).


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/oscarlab/graphene/1385)
